<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>If I understand correctly, coloc.abf() can be run with correlated variants, that is, no prerequisite for taking through LD pruning/clumping is required. Am I correct in my understanding ? • coloc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css"><script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet"><script src="pkgdown.js"></script><meta property="og:title" content="If I understand correctly, coloc.abf() can be run with correlated variants, that is, no prerequisite for taking through LD pruning/clumping is required. Am I correct in my understanding ?"><meta property="og:image" content="https://chr1swallace.github.io/coloc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-title-body">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">6.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="articles/a01_intro.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="articles/a02_data.html">Coloc: data structures</a>
    </li>
    <li>
      <a href="articles/a03_enumeration.html">Coloc: under a single causal variant assumption</a>
    </li>
    <li>
      <a href="articles/a04_sensitivity.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="articles/a06_SuSiE.html">Coloc: using SuSiE to relax the single causal variant assumption</a>
    </li>
    <li>
      <a href="articles/a07_variant-specific-priors.html">Coloc: using variant-specific priros in coloc</a>
    </li>
    <li>
      <a href="articles/a05_conditioning.html">DEPRECATED Coloc: relaxing the single causal variant assumption</a>
    </li>
  </ul></li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/chr1swallace/coloc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>If I understand correctly, coloc.abf() can be run with correlated variants, that is, no prerequisite for taking through LD pruning/clumping is required. Am I correct in my understanding ?</h1>
    </div>


<p><strong>Table of Contents</strong></p>
<ul><li><a href="#if-i-understand-correctly-colocabf-can-be-run-with-correlated-variants-that-is-no-prerequisite-for-taking-through-ld-pruningclumping-is-required-am-i-correct-in-my-understanding-">If I understand correctly, coloc.abf() can be run with correlated variants, that is, no prerequisite for taking through LD pruning/clumping is required. Am I correct in my understanding ?</a></li>
<li><a href="#assume-i-identify-a-sentinel-variant-for-a-block-of-genome-can-i-do-a-comparison-with-just-one-variant-using-colocabf">Assume I identify a sentinel variant for a block of genome, can I do a comparison with just one variant using coloc.abf()?</a></li>
<li><a href="#can-the-process-of-identifying-colocalized-variants-be-carried-out-genome-wide">Can the process of identifying colocalized variants be carried out genome wide?</a></li>
<li><a href="#how-is-colocabf-accounting-for-the-correlated-variants">How is coloc.abf accounting for the correlated variants?</a></li>
<li><a href="#how-to-define-priors-is-it-dependent-on-sample-size-or-any-other-parameters">How to define priors, is it dependent on sample size or any other parameters?</a></li>
<li><a href="#what-does-high-pp4-mean">What does high PP4 mean?</a></li>
</ul><!-- markdown-toc end --><div id="if-i-understand-correctly-colocabf-can-be-run-with-correlated-variants-that-is-no-prerequisite-for-taking-through-ld-pruningclumping-is-required-am-i-correct-in-my-understanding-" class="section level1">

<p>Yes, coloc.abf() and coloc.signals() assume they are given a dense map of all SNPs in a region that could be causal. Do not prune and clump.</p>
</div>
<div class="section level1">
<h1 id="assume-i-identify-a-sentinel-variant-for-a-block-of-genome-can-i-do-a-comparison-with-just-one-variant-using-colocabf">Assume I identify a sentinel variant for a block of genome, can I do a comparison with just one variant using coloc.abf()?<a class="anchor" aria-label="anchor" href="#assume-i-identify-a-sentinel-variant-for-a-block-of-genome-can-i-do-a-comparison-with-just-one-variant-using-colocabf"></a></h1>
<p>No, coloc.abf() and coloc.signals() assume they are given a dense map of all SNPs in a region that could be causal. This means you need to give all SNPs in a region. You can imagine they ask whether the patterns “match” across this region of SNPs, and a single variant does not represent a pattern.</p>
</div>
<div class="section level1">
<h1 id="can-the-process-of-identifying-colocalized-variants-be-carried-out-genome-wide">Can the process of identifying colocalized variants be carried out genome wide?<a class="anchor" aria-label="anchor" href="#can-the-process-of-identifying-colocalized-variants-be-carried-out-genome-wide"></a></h1>
<p>Coloc is designed to address whether two traits share causal variant(s) in a genomic region. It leaves the definition of “region” up to the user. You need to break the genome into smaller regions, within which it is reasonable to assume there is at most one (coloc.abf) or a small number (coloc.signals) of causal variants per trait. One way to do this is to use the boundaries defined by recombination hotspots, proxied by <a href="https://bitbucket.org/nygcresearch/ldetect-data/src/master/" class="external-link">this map</a> created by <a href="https://academic.oup.com/bioinformatics/article/32/2/283/1743626" class="external-link">lddetect</a>.</p>
<p>How big should a region be? Big enough that all variants in LD with a lead SNP are included; small enough that only one or a small number of causal variants might exist in it. I have found using densely genotyped studies and the lddetect boundaries above that regions typically contain 1,000-10,000 SNPs.</p>
</div>
<div class="section level1">
<h1 id="how-is-colocabf-accounting-for-the-correlated-variants">How is coloc.abf accounting for the correlated variants?<a class="anchor" aria-label="anchor" href="#how-is-colocabf-accounting-for-the-correlated-variants"></a></h1>
<p>That’s really how coloc works - by exploiting a dense SNP map - please see the <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383" class="external-link">original paper</a></p>
</div>
<div class="section level1">
<h1 id="how-to-define-priors-is-it-dependent-on-sample-size-or-any-other-parameters">How to define priors, is it dependent on sample size or any other parameters?<a class="anchor" aria-label="anchor" href="#how-to-define-priors-is-it-dependent-on-sample-size-or-any-other-parameters"></a></h1>
<p>This is described in detail in the <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1008720" class="external-link">latest paper</a></p>
</div>
<div class="section level1">
<h1 id="what-does-high-pp4-mean">What does high PP4 mean?<a class="anchor" aria-label="anchor" href="#what-does-high-pp4-mean"></a></h1>
<p>The summary printed on the screen by coloc.abf() and coloc.signals() shows the posterior probability of whether a shared causal variant exists in the region. High PP4 does not mean all variants are causal and shared - to check which variants are most likely to be causal look at the SNP.PP column in the returned detailed results data.frame.</p>
</div>
<div class="section level1">
<h1 id="i-have-a-question-how-should-we-interpret-a-high-overall-pp4-95-but-a-low-pp4-for-single-snp-50-in-colocabf-in-this-scenario-pp0-pp3-were-small-than-00001">I have a question. How should we interpret a high overall PP4 (&gt;95%) but a low PP4 for single SNP (&lt;50%) in coloc.abf? In this scenario, PP0-PP3 were small than 0.0001.<a class="anchor" aria-label="anchor" href="#i-have-a-question-how-should-we-interpret-a-high-overall-pp4-95-but-a-low-pp4-for-single-snp-50-in-colocabf-in-this-scenario-pp0-pp3-were-small-than-00001"></a></h1>
<p>high overall PP4 (&gt;95%) means there is a high probability of colocalisation</p>
<p>low PP4 for single SNP (&lt;50%) means we cannot identify which individual SNP is jointly causal with confidence</p>
</div>


  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by Chris Wallace, Claudia Giambartolomei, Tom Willis, Jeffrey Pullin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

