<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Colocalisation Tests of Two Genetic Traits • coloc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Colocalisation Tests of Two Genetic Traits">
<meta property="og:description" content="Performs the colocalisation tests described in Giambartolomei et al (2013) &lt;doi:10.1371/journal.pgen.1004383&gt;, Wallace (2020) &lt;doi:10.1371/journal.pgen.1008720&gt;, Wallace (2021) &lt;doi:10.1371/journal.pgen.1009440&gt;, Pullin and Wallace (2025) &lt;doi:10.1101/2024.08.21.608957&gt;.">
<meta property="og:image" content="https://chr1swallace.github.io/coloc/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">6.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/a01_intro.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="articles/a02_data.html">Coloc: data structures</a>
    </li>
    <li>
      <a href="articles/a03_enumeration.html">Coloc: under a single causal variant assumption</a>
    </li>
    <li>
      <a href="articles/a04_sensitivity.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="articles/a06_SuSiE.html">Coloc: using SuSiE to relax the single causal variant assumption</a>
    </li>
    <li>
      <a href="articles/a05_conditioning.html">DEPRECATED Coloc: relaxing the single causal variant assumption</a>
    </li>
    <li>
      <a href="articles/a07_variant-specific-priors.html">Coloc: using variant-specific priros in coloc</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/chr1swallace/coloc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="coloc">coloc<a class="anchor" aria-label="anchor" href="#coloc"></a>
</h1></div>
<p><a href="https://cran.r-project.org/package=coloc" class="external-link"><img src="https://www.r-pkg.org/badges/version/coloc" alt="CRAN_Status_Badge"></a></p>
<p><img src="reference/figures/logo30.png" align="right"> The coloc package can be used to perform genetic colocalisation analysis of two potentially related phenotypes, to ask whether they share common genetic causal variant(s) in a given region.</p>
<p>Most of the questions I get relate to misunderstanding the assumptions behind coloc (dense genotypes across a single genomic region) and/or the data structures used. Please read <code><a href="articles/a02_data.html">vignette("a02_data",package="coloc")</a></code> before starting an issue.</p>
<div class="section level2">
<h2 id="version-6">version 6<a class="anchor" aria-label="anchor" href="#version-6"></a>
</h2>
<p>This update (version 6) adds the ability to use variant-specific priors in coloc. See</p>
<ul>
<li>Pullin, J., Wallace, C. (2024+). Variant-specific priors in colocalisation analysis. bioRxiv 2024.08.21.608957. <a href="https://doi.org/10.1101/2024.08.21.608957" class="external-link uri">https://doi.org/10.1101/2024.08.21.608957</a>
</li>
</ul>
<p>for a description of the implementation and a comparison of different sources of prior inforamtion. If you use variant-specific priors in coloc please cite this paper.</p>
<p>More detail is also available in the vignette a07_variant_specific_priors.html accessible by</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"a07-variant_specific_priors"</span>,package<span class="op">=</span><span class="st">"coloc"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="version-5">version 5<a class="anchor" aria-label="anchor" href="#version-5"></a>
</h2>
<p>This update (version 5) supercedes previously published version 4 by introducing use of the <a href="https://stephenslab.github.io/susieR/index.html" class="external-link">SuSiE</a> approach to deal with multiple causal variants rather than conditioning or masking. See</p>
<ul>
<li>Wang, G., Sarkar, A., Carbonetto, P., &amp; Stephens, M. (2020). A simple new approach to variable selection in regression, with application to genetic fine mapping. Journal of the Royal Statistical Society: Series B (Statistical Methodology). <a href="https://doi.org/10.1111/rssb.12388" class="external-link uri">https://doi.org/10.1111/rssb.12388</a>
</li>
</ul>
<p>for the full SuSiE paper and</p>
<ul>
<li>Wallace (2021). A more accurate method for colocalisation analysis allowing for multiple causal variants. PLoS Genetics. <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1009440" class="external-link uri">https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1009440</a>
</li>
</ul>
<p>for a description of its use in coloc.</p>
<p>To install from R, do</p>
<pre><code><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://remotes.r-lib.org" class="external-link">"remotes"</a></span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> <span class="co"># if necessary</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">"chr1swallace/coloc@main"</span>,build_vignettes<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
<p>Note that in all simulations, susie outperforms the earlier conditioning approach, so is recommended. However, it is also new code, so please consider the code “beta” and let me know of any issues that arise - they may be a bug on my part. If you want to use it, the function you want to look at is <code>coloc.susie</code>. It can take raw datasets, but the time consuming part is running SuSiE. coloc runs SuSiE and saves a little extra information using the <code>runsusie</code> function before running an adapted colocalisation on the results. So please look at the docs for <code>runsusie</code> too. I found a helpful recipe is 1. Run <code>runsusie</code> on dataset 1, storing the results 2. Run <code>runsusie</code> on dataset 2, storing the results 3. Run <code>coloc.susie</code> on the two outputs from above</p>
<p>More detail is available in the vignette a06_SuSiE.html accessible by</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"a06_SuSiE"</span>,package<span class="op">=</span><span class="st">"coloc"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level1">
<h1 id="background-reading">Background reading<a class="anchor" aria-label="anchor" href="#background-reading"></a>
</h1>
<p>For usage, please see the vignette at <a href="https://chr1swallace.github.io/coloc" class="uri">https://chr1swallace.github.io/coloc</a></p>
<p>Key previous references are:</p>
<ul>
<li><p>original propostion of proportional colocalisation <a href="https://pubmed.ncbi.nlm.nih.gov/19039033/" class="external-link">Plagnol et al (2009)</a></p></li>
<li><p>proportional colocalisation with type 1 error rate control <a href="https://onlinelibrary.wiley.com/doi/abs/10.1002/gepi.21765" class="external-link">Wallace et al (2013)</a></p></li>
<li><p>colocalisation by enumerating all the possible causal SNP configurations between two traits, assuming at most one causal variant per trait <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383" class="external-link">Giambartolomei et al (2013)</a></p></li>
<li><p>Thoughts about priors in coloc are described in <a href="https://doi.org/10.1371/journal.pgen.1008720" class="external-link">Wallace C (2020) Eliciting priors and relaxing the single causal variant assumption in colocalisation analyses. PLOS Genetics 16(4): e1008720</a></p></li>
</ul>
</div>
<div class="section level1">
<h1 id="frequently-asked-questions">Frequently Asked Questions<a class="anchor" aria-label="anchor" href="#frequently-asked-questions"></a>
</h1>
<p>see <a href="https://chr1swallace.github.io/coloc/FAQ.html">FAQ</a></p>
</div>
<div class="section level1">
<h1 id="notes-to-self">Notes to self<a class="anchor" aria-label="anchor" href="#notes-to-self"></a>
</h1>
<!-- *to generate vignettes:* -->
<!-- ``` -->
<!-- cp vignettes/colocqq-tests.R.tospin vignettes/colocqq-tests.R && Rscript -e 'knitr::spin("vignettes/colocqq-tests.R",knit=FALSE); devtools::build_vignettes()' -->
<!-- ``` -->
<p><em>to generate website:</em> <a href="https://chr1swallace.github.io/coloc/" class="uri">https://chr1swallace.github.io/coloc/</a></p>
<pre><code>Rscript -e "pkgdown::build_site()"</code></pre>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=coloc" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/chr1swallace/coloc/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/chr1swallace/coloc/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GPL</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing coloc</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Chris Wallace <br><small class="roles"> Author, maintainer </small>  </li>
<li>Claudia Giambartolomei <br><small class="roles"> Author </small>  </li>
<li>Tom Willis <br><small class="roles"> Author </small>  </li>
<li>Jeffrey Pullin <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Chris Wallace, Claudia Giambartolomei, Tom Willis, Jeffrey Pullin.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
