<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Coloc: data structures • coloc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Coloc: data structures">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">6.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a01_intro.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="../articles/a02_data.html">Coloc: data structures</a>
    </li>
    <li>
      <a href="../articles/a03_enumeration.html">Coloc: under a single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a04_sensitivity.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="../articles/a06_SuSiE.html">Coloc: using SuSiE to relax the single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a07_variant-specific-priors.html">Coloc: using variant-specific priros in coloc</a>
    </li>
    <li>
      <a href="../articles/a05_conditioning.html">DEPRECATED Coloc: relaxing the single causal variant assumption</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/chr1swallace/coloc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Coloc: data structures</h1>
                        <h4 data-toc-skip class="author">Chris
Wallace</h4>
            
            <h4 data-toc-skip class="date">2025-03-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/chr1swallace/coloc/blob/HEAD/vignettes/a02_data.Rmd" class="external-link"><code>vignettes/a02_data.Rmd</code></a></small>
      <div class="hidden name"><code>a02_data.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="data-structures">Data structures<a class="anchor" aria-label="anchor" href="#data-structures"></a>
</h2>
<p>The enumeration approaches to coloc <a href="a03_enumeration.html"><code>coloc.abf</code></a> and <a href="a06_SuSiE.html"><code>coloc.susie</code></a> require the same data
format. To be flexible, <code>coloc.abf</code> allows a wider variety of
inputs, which can sometimes be confusing. This document aims to set out
what ideal and acceptable data structures are, as well as covering some
points on how you should select the SNPs for study.</p>
<div class="section level3">
<h3 id="first-select-the-snps-to-study-">First: select the SNPs to study.<a class="anchor" aria-label="anchor" href="#first-select-the-snps-to-study-"></a>
</h3>
<p>SNPs must * have summary data available in both studies * cover a
single genomic region (perhaps defined by distance about some GWAS peak
or delimited by recombination hotspots), and * represent a dense
coverage of the region</p>
<p>This can result in up to 10,000 SNPs on larger regions. A single
genomic region <em>does not</em> correspond to the whole genome, nor to
a whole chromosome. Coloc also does not automatically split chromosomes
or a genome into regions. It is assumed the user can look at their data,
identify a region with overlapping GWAS signals between two studies, and
decide on the set of SNPs to include. Some utilities are provided which
can help check regions are sensible.</p>
<p>Please do include all SNPs you have in the region: do not trim by
significance or MAF or other means. Coloc enumerates all possible
configureations of shared and non-shared causal variants and evaluates
their relative support; it cannot do this if it does not have all the
SNPs available.</p>
</div>
<div class="section level3">
<h3 id="a-minimum-coloc-dataset">A minimum coloc dataset<a class="anchor" aria-label="anchor" href="#a-minimum-coloc-dataset"></a>
</h3>
<p>Colocalisation and fine-mapping work through evaluating approximate
Bayes factors summarising evidence for association at each SNP. This
requires knowing the estimated effect size
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>
(<code>beta</code>) and its uncertainty
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>v</mi><mi>a</mi><mi>r</mi><mo stretchy="false" form="prefix">(</mo><mi>β</mi><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">var(\beta)</annotation></semantics></math>
(<code>varbeta</code>).
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>v</mi><mi>a</mi><mi>r</mi><mo stretchy="false" form="prefix">(</mo><mi>β</mi><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">var(\beta)</annotation></semantics></math>
is the standard error of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>,
squared. It is also good practice to supply SNP names
(<code>snp</code>), and SNP positions (<code>position</code>) are useful
if you want to use coloc’s <code><a href="../reference/plot_dataset.html">plot_dataset()</a></code>.</p>
<p>We also need a prior on the true effect sizes at causal variants,
which is different depending on whether you have a quantitative or
case-control (binary) trait. To estimate a prior, we need to know the
trait standard deviation (which allows us to interpret the scale of the
estimated beta).</p>
<p>A minimum coloc dataset is a <strong>list</strong>, containing a mix
of vectors (values that vary across snps, such as effect sizes
<code>beta</code> or variance of beta <code>varbeta</code>) and scalars
(values assumed to be constant across snps, such as sample size
[<code>N</code>] or data type [<code>type="cc"</code> for case control
or <code>type="quant"</code> for quantitative]). See
<code><a href="../reference/check_dataset.html">?check_dataset</a></code> for the full set of entries, some
combination of which are compulsory.</p>
<p>It can be useful to run <code><a href="../reference/check_dataset.html">check_dataset()</a></code> on each of your
datasets if coloc functions aren’t running. It will return NULL if it
finds no errors. It will stop with error if there are issues that
absolutely must be fixed to run coloc, but may print warnings if it sees
something merely suspicious - eg if there is no strong signal of
association.</p>
<p>Coloc comes with test data, so you can explore the structure for a
quantitative trait. Here we show the minimum dataset needed for a
quantitative trait.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/chr1swallace/coloc" class="external-link">coloc</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## This is coloc version 6.0.0</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">coloc_test_data</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/attach.html" class="external-link">attach</a></span><span class="op">(</span><span class="va">coloc_test_data</span><span class="op">)</span></span>
<span><span class="va">minimum_data</span><span class="op">=</span><span class="va">D1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"beta"</span>,<span class="st">"varbeta"</span>,<span class="st">"snp"</span>,<span class="st">"position"</span>,<span class="st">"type"</span>,<span class="st">"sdY"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">minimum_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 6</span></span>
<span><span class="co">##  $ beta    : num [1:500] 0.337 0.211 0.257 0.267 0.247 ...</span></span>
<span><span class="co">##  $ varbeta : num [1:500] 0.01634 0.00532 0.00748 0.01339 0.00664 ...</span></span>
<span><span class="co">##  $ snp     : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span>
<span><span class="co">##  $ position: int [1:500] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##  $ type    : chr "quant"</span></span>
<span><span class="co">##  $ sdY     : num 1.1</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check_dataset.html">check_dataset</a></span><span class="op">(</span><span class="va">minimum_data</span>,warn.minp<span class="op">=</span><span class="fl">1e-10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<p>Here we see (from <code>str</code>) that <code>minimum_data</code> is
a list. It has vector entries beta, varbeta, snp and position, all of
the same length (the number of snps in the region) and scalar entries
for type and sdY (the standard deviation of the quantitative trait).</p>
<p><code><a href="../reference/check_dataset.html">check_dataset()</a></code> issues a <strong>warning</strong> that
the minimum p value exceeds <code>warn.minp</code>. (The default value
for <code>warn.minp</code> is 10^{-6}). The warning doesn’t mean you
can’t run coloc - it is perfectly valid to assess that there is only
evidence for association in one of your datasets, but it may alert you
to check something if you <em>thought</em> there was strong association
there. One common mistake is to use the standard error of beta in place
of the variance of beta. If your dataset provides the standard error,
simply square it to get the variance.</p>
<p>It’s also useful to plot your datasets, and check * coverage is dense
* the level of association matches your expectation</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_dataset.html">plot_dataset</a></span><span class="op">(</span><span class="va">minimum_data</span><span class="op">)</span></span></code></pre></div>
<p><img src="a02_data_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<p>For a case control trait, you would need to give the
<code>type</code> as “cc”:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">minimum_ccdata</span><span class="op">=</span><span class="va">D1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"beta"</span>,<span class="st">"varbeta"</span>,<span class="st">"snp"</span>,<span class="st">"position"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">minimum_ccdata</span><span class="op">$</span><span class="va">type</span><span class="op">=</span><span class="st">"cc"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">minimum_ccdata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 5</span></span>
<span><span class="co">##  $ beta    : num [1:500] 0.337 0.211 0.257 0.267 0.247 ...</span></span>
<span><span class="co">##  $ varbeta : num [1:500] 0.01634 0.00532 0.00748 0.01339 0.00664 ...</span></span>
<span><span class="co">##  $ snp     : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span>
<span><span class="co">##  $ position: int [1:500] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##  $ type    : chr "cc"</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check_dataset.html">check_dataset</a></span><span class="op">(</span><span class="va">minimum_ccdata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<p>In either case, you can see that the lengths of beta, varbeta, snp
and position are the same. They are assumed to represent the same snps
in the same order. If you have your data in a data.frame, you could use
<code><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list()</a></code> to convert to a list, and then add the other
elements, but you cannot pass a data.frame to coloc, because elements
like <code>type</code> need to be of length 1.</p>
</div>
<div class="section level3">
<h3 id="analysing-two-datasets">Analysing two datasets<a class="anchor" aria-label="anchor" href="#analysing-two-datasets"></a>
</h3>
<p>coloc is used to analyse exactly two datasets. The snp identifiers
are used to match snps between datasets, and only the snps that appear
in both datasets will contribute to the analysis.</p>
</div>
<div class="section level3">
<h3 id="what-if-i-dont-have-sdy">What if I don’t have sdY?<a class="anchor" aria-label="anchor" href="#what-if-i-dont-have-sdy"></a>
</h3>
<p>Often, sdY is not available for quantitative datasets. If the study
say they standardised their trait to have variance 1, then you can set
=sdY= to 1. Otherwise coloc can estimate it, but needs MAF and sample
size information too. So the dataset would look like:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nosdY_data</span><span class="op">=</span><span class="va">D1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"beta"</span>,<span class="st">"varbeta"</span>,<span class="st">"snp"</span>,<span class="st">"position"</span>,<span class="st">"type"</span>,<span class="st">"N"</span>,<span class="st">"MAF"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">nosdY_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 7</span></span>
<span><span class="co">##  $ beta    : num [1:500] 0.337 0.211 0.257 0.267 0.247 ...</span></span>
<span><span class="co">##  $ varbeta : num [1:500] 0.01634 0.00532 0.00748 0.01339 0.00664 ...</span></span>
<span><span class="co">##  $ snp     : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span>
<span><span class="co">##  $ position: int [1:500] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##  $ type    : chr "quant"</span></span>
<span><span class="co">##  $ N       : int 1000</span></span>
<span><span class="co">##  $ MAF     : num [1:500] 0.031 0.166 0.0925 0.0405 0.118 ...</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check_dataset.html">check_dataset</a></span><span class="op">(</span><span class="va">nosdY_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="what-if-i-dont-have-beta-andor-varbeta">What if I don’t have beta and/or varbeta?<a class="anchor" aria-label="anchor" href="#what-if-i-dont-have-beta-andor-varbeta"></a>
</h3>
<p>beta and varbeta are preferred. Please supply them if you have them,
especially in the case of imputed data where they capture the effects of
imperfect imputation. But if you don’t have them, coloc can estimate
them, given p values, MAF, sample size and, if case-control data, the
fraction of samples that are cases:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nobeta_data</span><span class="op">=</span><span class="va">D1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAF"</span>,<span class="st">"snp"</span>,<span class="st">"position"</span>,<span class="st">"type"</span>,<span class="st">"sdY"</span>,<span class="st">"N"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">nobeta_data</span><span class="op">$</span><span class="va">pvalues</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">pnorm</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">D1</span><span class="op">$</span><span class="va">beta</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">D1</span><span class="op">$</span><span class="va">varbeta</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">nobeta_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 7</span></span>
<span><span class="co">##  $ MAF     : num [1:500] 0.031 0.166 0.0925 0.0405 0.118 ...</span></span>
<span><span class="co">##  $ snp     : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span>
<span><span class="co">##  $ position: int [1:500] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##  $ type    : chr "quant"</span></span>
<span><span class="co">##  $ sdY     : num 1.1</span></span>
<span><span class="co">##  $ N       : int 1000</span></span>
<span><span class="co">##  $ pvalues : num [1:500] 0.0083 0.00385 0.00294 0.02088 0.00242 ...</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check_dataset.html">check_dataset</a></span><span class="op">(</span><span class="va">nobeta_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nobeta_ccdata</span><span class="op">=</span><span class="va">D1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAF"</span>,<span class="st">"snp"</span>,<span class="st">"position"</span>,<span class="st">"N"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">nobeta_ccdata</span><span class="op">$</span><span class="va">pvalues</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">pnorm</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">D1</span><span class="op">$</span><span class="va">beta</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="va">D1</span><span class="op">$</span><span class="va">varbeta</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">*</span><span class="fl">2</span></span>
<span><span class="va">nobeta_ccdata</span><span class="op">$</span><span class="va">type</span><span class="op">=</span><span class="st">"cc"</span></span>
<span><span class="va">nobeta_ccdata</span><span class="op">$</span><span class="va">s</span><span class="op">=</span><span class="fl">0.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">nobeta_ccdata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 7</span></span>
<span><span class="co">##  $ MAF     : num [1:500] 0.031 0.166 0.0925 0.0405 0.118 ...</span></span>
<span><span class="co">##  $ snp     : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span>
<span><span class="co">##  $ position: int [1:500] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span><span class="co">##  $ N       : int 1000</span></span>
<span><span class="co">##  $ pvalues : num [1:500] 0.0083 0.00385 0.00294 0.02088 0.00242 ...</span></span>
<span><span class="co">##  $ type    : chr "cc"</span></span>
<span><span class="co">##  $ s       : num 0.5</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/check_dataset.html">check_dataset</a></span><span class="op">(</span><span class="va">nobeta_ccdata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<p>Note that coloc doesn’t mind if your dataset has more entries than it
minimally needs, it will use what it needs, preferring supplied
beta/varbeta and sdY over estimation, and leave the rest.</p>
</div>
<div class="section level3">
<h3 id="additional-ld-information-for-coloc-susie">Additional LD information for <code>coloc.susie</code><a class="anchor" aria-label="anchor" href="#additional-ld-information-for-coloc-susie"></a>
</h3>
<p><code>coloc.abf</code> makes the simplifying assumption that each
trait has at most one causal variant in the region under consideration.
This means it does not need to know about correlation (LD) between SNPs,
because no model under consideration has two or more SNPs in it.
<code>coloc.susie</code> extends the functionality by allowing more than
one causal variant, but this means it needs to know the LD between the
SNPs. You may be able to get this from the study data, but if not,
reference data such as 1000 Genomes can be used.</p>
<p>Our test data has an LD matrix already. This will be a square numeric
matrix of dimension equal to the number of SNPs, with dimnames
corresponding to the SNP ids.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">D1</span><span class="op">$</span><span class="va">LD</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  num [1:500, 1:500] 1 0.365 0.54 0.851 0.463 ...</span></span>
<span><span class="co">##  - attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">##   ..$ : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span>
<span><span class="co">##   ..$ : chr [1:500] "s1" "s2" "s3" "s4" ...</span></span></code></pre>
<div class="section level4">
<h4 id="nb-studies-from-different-populations">NB studies from different populations<a class="anchor" aria-label="anchor" href="#nb-studies-from-different-populations"></a>
</h4>
<p>One of the nice things about <code>coloc.abf</code> is that it does
not assume that LD structures are similar between studies, as long as
there is a dense map of SNPs across the region. The same thing still
applies to <code>coloc.susie</code>, just add an appropriate LD matrix
to each study.</p>
</div>
<div class="section level4">
<h4 id="nb-allele-order-matters">NB allele order matters!<a class="anchor" aria-label="anchor" href="#nb-allele-order-matters"></a>
</h4>
<p>beta coefficients in GWAS studies report the effect of one allele
with respect to the other. That is, if you see results looking like</p>
<table class="table">
<thead><tr class="header">
<th>CHR38</th>
<th>BP38</th>
<th>REF</th>
<th>ALT</th>
<th>SNPID</th>
<th>P</th>
<th>BETA</th>
<th>SE</th>
<th>maf</th>
<th>maf_cases</th>
<th>maf_controls</th>
<th></th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>2101817</td>
<td>G</td>
<td>A</td>
<td>rs908742</td>
<td>0.0826</td>
<td>0.0749</td>
<td>0.0432</td>
<td>0.4394</td>
<td>0.456</td>
<td>0.4391</td>
<td></td>
</tr>
<tr class="even">
<td>1</td>
<td>2109497</td>
<td>T</td>
<td>C</td>
<td>rs4648808</td>
<td>0.8695</td>
<td>-0.0144</td>
<td>0.0877</td>
<td>0.9347</td>
<td>0.9344</td>
<td>0.9347</td>
<td></td>
</tr>
<tr class="odd">
<td>1</td>
<td>2126584</td>
<td>A</td>
<td>G</td>
<td>rs3128291</td>
<td>0.7334</td>
<td>-0.0306</td>
<td>0.0897</td>
<td>0.939</td>
<td>0.9371</td>
<td>0.9391</td>
<td></td>
</tr>
<tr class="even">
<td>1</td>
<td>2137467</td>
<td>G</td>
<td>T</td>
<td>rs3128296</td>
<td>0.8802</td>
<td>-0.0124</td>
<td>0.0826</td>
<td>0.9268</td>
<td>0.9267</td>
<td>0.9268</td>
<td></td>
</tr>
<tr class="odd">
<td>1</td>
<td>2139901</td>
<td>C</td>
<td>A</td>
<td>rs424079</td>
<td>0.6399</td>
<td>-0.0207</td>
<td>0.0442</td>
<td>0.6347</td>
<td>0.6269</td>
<td>0.6348</td>
<td></td>
</tr>
</tbody>
</table>
<p>the beta is telling you about the effect of allele A compared to
allele G at the snp rs908742. If the alleles were reversed, beta would
be -0.0826. The LD matrix needs to be aligned the same way. This will be
easy if you have the GWAS data and are making the LD matrix yourself. If
you get it from a public source, it will almost certainly be telling you
about the LD between non-reference alleles. In this case, you need to
make sure your data is also listing the non-reference allele as the
“alternative” or “risk” allele. If it is not, switch the sign of
beta.</p>
<p>To avoid hassles, I recommend: * if using GWAS catalog data, download
the <a href="https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics" class="external-link"><em>harmonised</em>
version</a> which will do this for you. * if using eQTL summary data
that has been processed by the <a href="https://www.ebi.ac.uk/gwas/docs/methods/summary-statistics" class="external-link">eQTL
catalogue</a>, use that version because it will also be harmonised *
Otherwise, check the alleles in your summary table match those in the
information files from the LD source * Finally, always plot the coloc
result with the <a href="a04_sensitivity.html"><code>sensitivity</code>
function</a> because weird effects are much easier to understand
visually</p>
<p>You can also use the coloc function <code>check_alignment</code> as a
visual guide. It compares the product of the Z scores for all SNP pairs
against their correlation, and plots a histogram of this ratio for SNPs
in some degree of LD. We do not expect to see strongly negative products
of Z scores for strongly positive correlations or vice versa. Therefore
this plot <em>should be</em> skewed to have more positive values than
negative. Here I show a good (left) and bad (right) example.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/check_alignment.html">check_alignment</a></span><span class="op">(</span><span class="va">goodD</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.9061496</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="fl">40</span>,<span class="fl">200</span>,<span class="st">"good"</span>,col<span class="op">=</span><span class="st">"red"</span>,cex<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/check_alignment.html">check_alignment</a></span><span class="op">(</span><span class="va">badD</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.4999569</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">200</span>,<span class="st">"bad"</span>,col<span class="op">=</span><span class="st">"red"</span>,cex<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="a02_data_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Chris Wallace, Claudia Giambartolomei, Tom Willis, Jeffrey Pullin.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
