<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index â€¢ coloc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="https://chr1swallace.github.io/coloc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">6.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a01_intro.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="../articles/a02_data.html">Coloc: data structures</a>
    </li>
    <li>
      <a href="../articles/a03_enumeration.html">Coloc: under a single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a04_sensitivity.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="../articles/a06_SuSiE.html">Coloc: using SuSiE to relax the single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a07_variant-specific-priors.html">Coloc: using variant-specific priros in coloc</a>
    </li>
    <li>
      <a href="../articles/a05_conditioning.html">DEPRECATED Coloc: relaxing the single causal variant assumption</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/chr1swallace/coloc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="coloc-data-structure">Coloc data structure <a href="#coloc-data-structure" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Description of coloc data structure</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="check_dataset.html">check_dataset()</a></code> <code><a href="check_dataset.html">check.dataset()</a></code> </p>
        </td>
        <td><p>check_dataset</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="bayesian-enumeration-of-hypotheses">Bayesian enumeration of hypotheses <a href="#bayesian-enumeration-of-hypotheses" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Two trait colocalisation and single trait fine mapping under single causal variant assumptions</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="coloc.abf.html">coloc.abf()</a></code> </p>
        </td>
        <td><p>Fully Bayesian colocalisation analysis using Bayes Factors</p></td>
      </tr><tr><td>
          <p><code><a href="finemap.abf.html">finemap.abf()</a></code> </p>
        </td>
        <td><p>Bayesian finemapping analysis</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="sensitivity-analysis">Sensitivity analysis <a href="#sensitivity-analysis" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Check how robust inference is to changing prior parameter values</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="sensitivity.html">sensitivity()</a></code> </p>
        </td>
        <td><p>Prior sensitivity for coloc</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="relaxing-the-single-variant-assumption">Relaxing the single variant assumption <a href="#relaxing-the-single-variant-assumption" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>use conditioning or masking to account for multiple independent causal variants per trait</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="coloc.susie.html">coloc.susie()</a></code> </p>
        </td>
        <td><p>run coloc using susie to detect separate signals</p></td>
      </tr><tr><td>
          <p><code><a href="runsusie.html">runsusie()</a></code> </p>
        </td>
        <td><p>Run susie on a single coloc-structured dataset</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="visualisation">Visualisation <a href="#visualisation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>see stuff</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="plot_dataset.html">plot_dataset()</a></code> </p>
        </td>
        <td><p>plot a coloc dataset</p></td>
      </tr><tr><td>
          <p><code><a href="plot_extended_dataset.html">plot_extended_dataset()</a></code> </p>
        </td>
        <td><p>Draw extended plot of summary statistics for two coloc datasets</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="others">Others <a href="#others" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>included for completeness, but the main functions you need should be in the sections above</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="annotate_susie.html">annotate_susie()</a></code> </p>
        </td>
        <td><p>annotate susie_rss output for use with coloc_susie</p></td>
      </tr><tr><td>
          <p><code><a href="approx.bf.estimates.html">approx.bf.estimates()</a></code> </p>
        </td>
        <td><p>Internal function, approx.bf.estimates</p></td>
      </tr><tr><td>
          <p><code><a href="approx.bf.p.html">approx.bf.p()</a></code> </p>
        </td>
        <td><p>Internal function, approx.bf.p</p></td>
      </tr><tr><td>
          <p><code><a href="bin2lin.html">bin2lin()</a></code> </p>
        </td>
        <td><p>binomial to linear regression conversion</p></td>
      </tr><tr><td>
          <p><code><a href="check_alignment.html">check_alignment()</a></code> <code><a href="check_alignment.html">check.alignment()</a></code> </p>
        </td>
        <td><p>check alignment</p></td>
      </tr><tr><td>
          <p><code><a href="coloc-package.html">-package</a></code> <code><a href="coloc-package.html">coloc-package</a></code> </p>
        </td>
        <td><p>Colocalisation tests of two genetic traits</p></td>
      </tr><tr><td>
          <p><code><a href="coloc.bf_bf.html">coloc.bf_bf()</a></code> </p>
        </td>
        <td><p>Coloc data through Bayes factors</p></td>
      </tr><tr><td>
          <p><code><a href="coloc.detail.html">coloc.detail()</a></code> </p>
        </td>
        <td><p>Bayesian colocalisation analysis with detailed output</p></td>
      </tr><tr><td>
          <p><code><a href="coloc.process.html">coloc.process()</a></code> </p>
        </td>
        <td><p>Post process a coloc.details result using masking</p></td>
      </tr><tr><td>
          <p><code><a href="coloc.signals.html">coloc.signals()</a></code> </p>
        </td>
        <td><p>Coloc with multiple signals per trait</p></td>
      </tr><tr><td>
          <p><code><a href="coloc.susie_bf.html">coloc.susie_bf()</a></code> </p>
        </td>
        <td><p>run coloc using susie to detect separate signals</p></td>
      </tr><tr><td>
          <p><code><a href="coloc_test_data.html">coloc_test_data</a></code> </p>
        </td>
        <td><p>Simulated data to use in testing and vignettes in the coloc package</p></td>
      </tr><tr><td>
          <p><code><a href="combine.abf.html">combine.abf()</a></code> </p>
        </td>
        <td><p>combine.abf</p></td>
      </tr><tr><td>
          <p><code><a href="credible.sets.html">credible.sets()</a></code> </p>
        </td>
        <td><p>credible.sets</p></td>
      </tr><tr><td>
          <p><code><a href="est_cond.html">est_cond()</a></code> </p>
        </td>
        <td><p>generate conditional summary stats</p></td>
      </tr><tr><td>
          <p><code><a href="estgeno1.html">estgeno.1.ctl()</a></code> <code><a href="estgeno1.html">estgeno.1.cse()</a></code> </p>
        </td>
        <td><p>estgeno1</p></td>
      </tr><tr><td>
          <p><code><a href="find.best.signal.html">find.best.signal()</a></code> </p>
        </td>
        <td><p>Pick out snp with most extreme Z score</p></td>
      </tr><tr><td>
          <p><code><a href="findends.html">findends()</a></code> </p>
        </td>
        <td><p>trim a dataset to central peak(s)</p></td>
      </tr><tr><td>
          <p><code><a href="findpeaks.html">findpeaks()</a></code> </p>
        </td>
        <td><p>trim a dataset to only peak(s)</p></td>
      </tr><tr><td>
          <p><code><a href="finemap.bf.html">finemap.bf()</a></code> </p>
        </td>
        <td><p>Finemap data through Bayes factors</p></td>
      </tr><tr><td>
          <p><code><a href="finemap.signals.html">finemap.signals()</a></code> </p>
        </td>
        <td><p>Finemap multiple signals in a single dataset</p></td>
      </tr><tr><td>
          <p><code><a href="logbf_to_pp.html">logbf_to_pp()</a></code> </p>
        </td>
        <td><p>logbf 2 pp</p></td>
      </tr><tr><td>
          <p><code><a href="logdiff.html">logdiff()</a></code> </p>
        </td>
        <td><p>logdiff</p></td>
      </tr><tr><td>
          <p><code><a href="logsum.html">logsum()</a></code> </p>
        </td>
        <td><p>logsum</p></td>
      </tr><tr><td>
          <p><code><a href="map_cond.html">map_cond()</a></code> </p>
        </td>
        <td><p>find the next most significant SNP, conditioning on a list of sigsnps</p></td>
      </tr><tr><td>
          <p><code><a href="map_mask.html">map_mask()</a></code> </p>
        </td>
        <td><p>find the next most significant SNP, masking a list of sigsnps</p></td>
      </tr><tr><td>
          <p><code><a href="plot-methods.html">plot(<i>&lt;coloc_abf&gt;</i>)</a></code> </p>
        </td>
        <td><p>plot a coloc_abf object</p></td>
      </tr><tr><td>
          <p><code><a href="print-methods.html">print(<i>&lt;coloc_abf&gt;</i>)</a></code> </p>
        </td>
        <td><p>print.coloc_abf</p></td>
      </tr><tr><td>
          <p><code><a href="process.dataset.html">process.dataset()</a></code> </p>
        </td>
        <td><p>process.dataset</p></td>
      </tr><tr><td>
          <p><code><a href="sdY.est.html">sdY.est()</a></code> </p>
        </td>
        <td><p>Estimate trait variance, internal function</p></td>
      </tr><tr><td>
          <p><code><a href="subset_dataset.html">subset_dataset()</a></code> </p>
        </td>
        <td><p>subset_dataset</p></td>
      </tr><tr><td>
          <p><code><a href="Var.data.cc.html">Var.data.cc()</a></code> </p>
        </td>
        <td><p>Var.data</p></td>
      </tr><tr><td>
          <p><code><a href="Var.data.html">Var.data()</a></code> </p>
        </td>
        <td><p>Var.data</p></td>
      </tr><tr><td>
          <p><code><a href="eqtlgen_density_data.html">eqtlgen_density_data</a></code> </p>
        </td>
        <td><p>eQTLGen estimated distance density</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Chris Wallace, Claudia Giambartolomei, Tom Willis, Jeffrey Pullin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

