<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Draw extended plot of summary statistics for two coloc datasets — plot_extended_dataset • coloc</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Draw extended plot of summary statistics for two coloc datasets — plot_extended_dataset"><meta property="og:description" content="Draw Manhattan-style locus plots for p-values in each dataset, gene annotations,
a scatter plot of z-scores, and a table of coloc summary statistics."><meta property="og:image" content="https://chr1swallace.github.io/coloc/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">coloc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">6.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/a01_intro.html">Coloc: a package for colocalisation analyses</a>
    </li>
    <li>
      <a href="../articles/a02_data.html">Coloc: data structures</a>
    </li>
    <li>
      <a href="../articles/a03_enumeration.html">Coloc: under a single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a04_sensitivity.html">Coloc: sensitivity to prior values</a>
    </li>
    <li>
      <a href="../articles/a06_SuSiE.html">Coloc: using SuSiE to relax the single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a05_conditioning.html">DEPRECATED Coloc: relaxing the single causal variant assumption</a>
    </li>
    <li>
      <a href="../articles/a07_variant-specific-priors.html">Coloc: using variant-specific priros in coloc</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/chr1swallace/coloc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Draw extended plot of summary statistics for two coloc datasets</h1>
    <small class="dont-index">Source: <a href="https://github.com/chr1swallace/coloc/blob/HEAD/R/plot.R" class="external-link"><code>R/plot.R</code></a></small>
    <div class="hidden name"><code>plot_extended_dataset.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Draw Manhattan-style locus plots for p-values in each dataset, gene annotations,
a scatter plot of z-scores, and a table of coloc summary statistics.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plot_extended_dataset</span><span class="op">(</span></span>
<span>  <span class="va">dataset1</span>,</span>
<span>  <span class="va">dataset2</span>,</span>
<span>  <span class="va">x</span>,</span>
<span>  first_highlight_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  second_highlight_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  first_index_snp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  second_index_snp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  first_trait <span class="op">=</span> <span class="st">"Trait 1"</span>,</span>
<span>  second_trait <span class="op">=</span> <span class="st">"Trait 2"</span>,</span>
<span>  snp_label <span class="op">=</span> <span class="st">"snp"</span>,</span>
<span>  ld_label <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_ld <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  locus_plot_ylim <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ens_db <span class="op">=</span> <span class="st">"EnsDb.Hsapiens.v86"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>object of class <code>coloc_abf</code> returned by <code><a href="coloc.abf.html">coloc.abf()</a></code></p></dd>


<dt id="arg-first-highlight-list">first_highlight_list<a class="anchor" aria-label="anchor" href="#arg-first-highlight-list"></a></dt>
<dd><p>character vector of snps to highlight in the first dataset;
<code>'index'</code> can be passed to highlight the most significant SNP</p></dd>


<dt id="arg-second-highlight-list">second_highlight_list<a class="anchor" aria-label="anchor" href="#arg-second-highlight-list"></a></dt>
<dd><p>character vector of snps to highlight in the second dataset;
<code>'index'</code> can be passed to highlight the most significant SNP</p></dd>


<dt id="arg-first-index-snp">first_index_snp<a class="anchor" aria-label="anchor" href="#arg-first-index-snp"></a></dt>
<dd><p>snp to designate as index snp in the first dataset
for the purpose of LD visualisation</p></dd>


<dt id="arg-second-index-snp">second_index_snp<a class="anchor" aria-label="anchor" href="#arg-second-index-snp"></a></dt>
<dd><p>snp to designate as index snp in the second dataset
for the purpose of LD visualisation</p></dd>


<dt id="arg-first-trait">first_trait<a class="anchor" aria-label="anchor" href="#arg-first-trait"></a></dt>
<dd><p>label for the first trait</p></dd>


<dt id="arg-second-trait">second_trait<a class="anchor" aria-label="anchor" href="#arg-second-trait"></a></dt>
<dd><p>label for the second trait</p></dd>


<dt id="arg-snp-label">snp_label<a class="anchor" aria-label="anchor" href="#arg-snp-label"></a></dt>
<dd><p>label for the snp column</p></dd>


<dt id="arg-ld-label">ld_label<a class="anchor" aria-label="anchor" href="#arg-ld-label"></a></dt>
<dd><p>label for the LD column</p></dd>


<dt id="arg-show-ld">show_ld<a class="anchor" aria-label="anchor" href="#arg-show-ld"></a></dt>
<dd><p>logical, whether to show LD in the locus plots</p></dd>


<dt id="arg-locus-plot-ylim">locus_plot_ylim<a class="anchor" aria-label="anchor" href="#arg-locus-plot-ylim"></a></dt>
<dd><p>numeric vector of length 2 specifying the y-axis
limits for the locus plots on the -log10 scale</p></dd>


<dt id="arg-ens-db">ens_db<a class="anchor" aria-label="anchor" href="#arg-ens-db"></a></dt>
<dd><p>character string specifying Ensembl database package from which to get gene positions. Current (at time of writing this documentation!) sensible values are  "EnsDb.Hsapiens.v86" for build 38 and "EnsDb.Hsapiens.v75" for build 37.</p></dd>


<dt id="arg-d">d<a class="anchor" aria-label="anchor" href="#arg-d"></a></dt>
<dd><p>a coloc dataset</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a gtable object</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Plot a pair of coloc datasets in an extended format</p>
<p>This function expects that the first two elements of the coloc
dataset list d contain summary statistics. Columns labelled 'chromosome',
'position', and 'p' (p-values) are expected in each dataset. The packages <code>locuszoomr</code>
and one of <code>EnsDb.Hsapiens.v86</code> or <code>EnsDb.Hsapiens.v75</code> are required
for this function. One of the <code>EnsDb.Hsapiens</code> libraries containing the Ensembl database
specified by ens_db must be loaded prior to  use (see the example). EnsDb.Hsapiens.v86
is the default database (GRCh38/hg19); for GRCh37/hg19, use EnsDb.Hsapiens.v75.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Tom Willis</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/chr1swallace/coloc" class="external-link">coloc</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu">plot_extended_dataset</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">first_dataset</span>, <span class="va">second_dataset</span><span class="op">)</span>, <span class="va">coloc</span>,</span></span>
<span class="r-in"><span>first_highlight_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs123"</span>, <span class="st">"rs456"</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>second_highlight_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs789"</span>, <span class="st">"rs1011"</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>ens_db <span class="op">=</span> <span class="st">"EnsDb.Hsapiens.v86"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Chris Wallace, Claudia Giambartolomei, Tom Willis, Jeffrey Pullin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

